src/llm/physical_optimizer.rs

```diff
@@ -0,0 +1,148 @@
+use crate::llm::async_func::AsyncFuncExpr;
+use crate::llm::exec::AsyncFuncExec;
+use datafusion::common::tree_node::{Transformed, TreeNode, TreeNodeRecursion};
+use datafusion::config::ConfigOptions;
+use datafusion::physical_expr::expressions::Column;
+use datafusion::physical_expr::{PhysicalExpr, ScalarFunctionExpr};
+use datafusion::physical_optimizer::PhysicalOptimizerRule;
+use datafusion::physical_plan::projection::ProjectionExec;
+use datafusion::physical_plan::ExecutionPlan;
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub struct AsyncFuncRule {}
+
+impl PhysicalOptimizerRule for AsyncFuncRule {
+    /// Insert a AsyncFunctionNode node in front of this projection if there are any async functions in it
```

Comment:
Here is the high level design: add a new node before a ProjectionExec that does the actual async calls